src/robot/api/exceptions.py

```diff
@@ -31,7 +31,7 @@ class Failure(AssertionError):
     """
     ROBOT_SUPPRESS_NAME = True
 
-    def __init__(self, message, html=False):
+    def __init__(self, message: str, html: bool = False) -> None:
```

Comment:
Please remove `-> None` from here and everywhere else. We can consider adding it to public APIs if we at some point aim for 100% type coverage, but at the moment they just add distraction in general and are totally useless with `__init__`s.

Reply:
Removing them from the `__init__` does not result in type-checking errors (a special case in mypy, actually), but removing the `-> None` annotations from the functions in `logger` and `interfaces` introduces 35 mypy errors (`[no-untyped-def]`). This is with `disallow_untyped_defs = true` OR `strict=true` in the mypy settings.

I personally feel we should aim for 100% annotations on external APIs. If we want to support users using type-checkers on their codebase, we need to be complete even if it's of little value for ourselves / within the robot codebase itself.

I've removed the return types on the `__init__`s and I'll remove the `-> None` returns if that's the direction / decision for now.